js/canvas.ts

```diff
@@ -356,24 +328,20 @@ export class LoresPage2D implements LoresPage {
                     }
                 }
             } else {
-                if (!_80colMode && bank == 1) {
-                    return;
-                }
-                if (_80colMode && !an3) {
+                if (this.vm._80colMode && !this.vm.an3State) {
                     let offset = (col * 14 + (bank ? 0 : 1) * 7 + row * 560 * 8) * 4;
-                    if (this._monoMode) {
-                        fore = whiteCol;
-                        back = blackCol;
+                    if (this.vm._monoMode) {
                         for (let jdx = 0; jdx < 8; jdx++) {
-                            let b = (jdx < 8) ? (val & 0x0f) : (val >> 4);
+                            let b = (jdx < 4) ? (val & 0x0f) : (val >> 4);
```

Comment:
Could you describe what's going on here with a comment? I'm having a hard time following it.

Reply:
It would probably be good to document all the various drawing modes - this was just broken (yay tests) and I copied the working mono version from the gl side. That said, I'd like to wait on that until I've done the next part of the refactor and eliminated duplicate code.